lib/parser/open_element_stack.js

```diff
@@ -278,12 +277,13 @@ OpenElementStack.prototype.hasInScope = function (tagName) {
 
 OpenElementStack.prototype.hasNumberedHeaderInScope = function () {
     for (var i = this.stackTop; i >= 0; i--) {
-        var tn = this.treeAdapter.getTagName(this.items[i]);
+        var tn = this.treeAdapter.getTagName(this.items[i]),
+            ns = this.treeAdapter.getNamespaceURI(this.items[i]);
 
-        if (tn === $.H1 || tn === $.H2 || tn === $.H3 || tn === $.H4 || tn === $.H5 || tn === $.H6)
+        if ((tn === $.H1 || tn === $.H2 || tn === $.H3 || tn === $.H4 || tn === $.H5 || tn === $.H6) && ns === NS.HTML)
```

Comment:
> elements in stack that are not headers 

If elements are not headers we will need namespace anyway to perform scoping element test: https://github.com/inikulin/parse5/pull/146/files#diff-c84848d80bdda10576c7ee69b4b5f652R295